Sources/DeepLearning/Layer.swift

```diff
@@ -52,12 +52,12 @@ public extension Layer {
 /// between training and inference shares an instance of ModeRef so it doesn't
 /// need to be toggled or threaded through in more than one place.
 public class ModeRef {
```

Comment:
Avoid using short names unless it's a word of art. In this case, this should be renamed to something like `InferenceModeIndicator`.

Reply:
I think I’m going to adopt the Keras terminology (“learning phase”), although I also like “mode” as a noun

Reply:
+1 to "learning phase", but I would add a word like "indicator" to indicate its reference semantics.